Add support for PHP 8

sebdesign opened the pull request.

winzou commented on .travis.yml:
```diff
@@ -7,6 +7,9 @@ language: php
     - 7.2
+    - 7.3
+    - 7.4
+    - nightly
```
I wouldn't add nightly here, or at least define it as allowed failure?

sebdesign replied:
I think `nightly` means php-8.0.0-RCx for the moment because it's not a stable version yet. We can add `- 8.0` once it's available in Travis and define `nightly` as allowed failures then. What do you think?

winzou commented on composer.json:
```diff
@@ -22,7 +22,7 @@
     },
     "require-dev": {
-        "phpspec/phpspec": "^5.0|^6.0",
+        "phpspec/phpspec": "^5.0|^6.3",
```
Why so?

sebdesign replied:
[v6.3.0](https://github.com/phpspec/phpspec/releases/tag/6.3.0) allows PHP 8, while keeping backwards compatibility for PHP >= 7.2.

winzou reviewed (commented):
Thanks for your PR! Can we check my comments before I merge?

sebdesign commented:
Let me know if you want me to change anything or if you have better suggestions. Merci!

sebdesign commented:
Hello Alexandre,

now that PHP 8 is released, I changed .travis.yml but is seems that they only have PHP 8 in `nightly`. I also reverted the phpspec dependency.

Let me know if there is something I can do to make this happen.

Thanks!

idsulik commented:
I'm waiting for it

nolka commented:
Is the some news about this pr?

idsulik commented:
@winzou , are you here?)

TeNNoX commented:
\*shy +1*

ABartelt commented:
@winzou how can we help you?

winzou commented:
Thanks for your PR! And sorry for the delay.
I've tagged version 0.4.1: https://github.com/winzou/state-machine/releases/tag/0.4.1

sebdesign commented:
Thank you very much @winzou !
No worries at all for the delay, I hope you're doing well.
